docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+KubeStellar users can have different roles and responsibilities. 
+For example, a System Admin can set up a KubeStellar instance, 
+create WDSs and ITSs, and register WECs. An Application Owner can 
+define Placement Policies and submit Workload Definitions for deployment. 
+Other roles, such as CISO or DevOps Engineers, may also exist. However, 
+in this document, we will not make an explicit distinction between
+different user roles and will refer to a user that has all these roles.
+The user interaction with KubeStellar is described in the 
+[user interaction](./user-interaction.md) section.
 
-The status addon defines the `WorkStatus` Kind of objects and some (what?) associated stuff.
+The KubeStellar architecture has these main modules:
 
-## KubeStellar
+- *KubeStellar Controller Manager*: this module is responsible of delivering workload
+objects from the WDS to the ITS according to placement policies, and 
+updating the status of objects in the WDS.
 
-KubeStellar currently has two major parts.
+- *Space Manager*: This module manages the lifecycle of spaces.
 
-1. OCM for inventory and transport
-2. A central controller-manager that provides the added functionalities
+- *OCM Cluster Manager*: This module syncs objects from the ITS to the Workload Execution 
+Clusters (WECs). In the ITS, each mailbox namespace is associated with one WEC. Objects 
+that are put in a mailbox namespace are delivered to the matching WEC.
 
-We plan to make the transport pluggable in the near future.
+- *Status Add-On Controller*: This module installs the OCM status add-on agent 
+on all WECs and sets the RBAC permissions for it using the OCM add-on framework.
 
+- *OCM Agent*: This module registers the WEC to the OCM Hub, watches for 
+ManifestWorks and unwraps and syncs the objects into the WEC.
+
+- *OCM Status Add-On Agent*: This module watches *AppliedManifestWorks* 
+to find objects that are synced by the OCM agent, gets their status 
+and updates *workstatuses* objects in the ITS namespace associated with the WEC.
+
+<figure>
+  <img src="assets/main-modules.png"  alt="Main Modules">
+  <figcaption align="center">Figure 2 - Main Modules </figcaption>
+</figure>
+
+## KubeStellar Controller Manager
+
+This module manages the placement and status controllers. The placement controller watches 
+placement and workload objects on the Workload Definition Space 
+(WDS) and wraps workload objects into a manifest for delivery through the 
+Inventory and Transport Space (ITS). The status controller watches for `workstatuses` objects on the 
+ITS and updates the status of objects in the WDS when singleton status is requested 
+in the placement for those objects. There is one instance of a KubeStellar Controller 
+Manager for each WDS. Currently this controller-manager runs in the KubeFlex hosting cluster.
+More details on the internals of this module are provided in [KubeStellar Controllers Architecture](#kubestellar-controllers-architecture)
+
+## Space Manager
+
```

Comment:
In general there are sections here that go into explaining what Kflex is , we need to think what should be included here and what should be simply pointed to the KF documentation

Reply:
I will leave the minimum that is needed for readers concerned with understanding KubeStellar overall arch and modules and move/point to KS docs for more details.